src/vs/workbench/contrib/chat/browser/contrib/chatInputEditorContrib.ts

```diff
@@ -131,7 +132,43 @@ class InputEditorDecorations extends Disposable {
 	}
 }
 
-ChatWidget.CONTRIBS.push(InputEditorDecorations);
+class InputEditorSlashCommandFollowups extends Disposable {
+	private slashCommands: Promise<ISlashCommand[] | undefined> | undefined;
+
+	constructor(
+		private readonly widget: IChatWidget,
+		@IChatService private readonly chatService: IChatService
+	) {
+		super();
+		this._register(this.chatService.onDidCompleteSlashCommand(({ slashCommand, sessionId }) => this.repopulateSlashCommand(slashCommand, sessionId)));
+	}
+
+	private getSlashCommands() {
+		if (!this.slashCommands && this.widget.viewModel?.sessionId) {
+			this.slashCommands = this.chatService.getSlashCommands(this.widget.viewModel?.sessionId, new CancellationTokenSource().token);
+		}
+
+		return this.slashCommands;
+	}
+
+	private async repopulateSlashCommand(slashCommand: string, sessionId: string) {
+		if (this.widget.viewModel?.sessionId !== sessionId) {
+			return;
+		}
+
+		const slashCommands = await this.getSlashCommands();
```

Comment:
You could just skip this step and trust that the slashCommand still exists, to avoid a delay before it shows up again. I really don't like these being async, it leads to some weird issues, and I don't think we have any scenario where they will disappear here.

Reply:
Or you could call `getSlashCommands` on the chat widget, that is there so the widget can decide when to invalidate cached slash commands